helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -77,7 +88,23 @@ public synchronized boolean addMetadataStoreRealm(String realm) {
       return createZkRealm(realm);
     }
 
-    // TODO: Forward the request to leader
+    String leaderHostName = _leaderElection.getCurrentLeaderInfo().getId();
+    String url =
+        leaderHostName + "/admin/v2/namespaces/" + _namespace + "/metadata-store-realms/" + realm;
+    HttpPut putRequest = new HttpPut(url);
+    try {
+      HttpResponse response = _forwardHttpClient.execute(putRequest);
+      if (response.getStatusLine().getStatusCode() != 201) {
+        HttpEntity respEntity = response.getEntity();
+        LOG.error("the forwarded request to leader has failed for addMetadataStoreRealm: {}",
+            respEntity != null ? EntityUtils.toString(respEntity) : "");
+        return false;
+      }
+    } catch (IOException e) {
+      LOG.error("the forwarded request to leader raised an exception for addMetadataStoreRealm", e);
```

Comment:
More helpful to include which host is forwarding to which leader.

Reply:
As in the hostnames? Okay.

Reply:
Like this:

`String errorMessage = "The forwarded request from " + (this node's hostname) + " to leader " + (leader's hostname) + " has failed for addMetadataStoreRealm()!";
if (respEntity != null) {
  errorMessage += " Response: " + EntityUtils.toString(respEntity);
}
LOG. error(errorMessage);`